src/openms/source/ANALYSIS/OPENSWATH/SpectrumExtractor.cpp

```diff
@@ -298,4 +300,60 @@ namespace OpenMS
     }
     LOG_DEBUG << "the annotated variable has " << annotated_spectra.size() << " elements instead of " << spectra.size() << std::endl;
   }
+
+  void SpectrumExtractor::scoreSpectrum(
+    std::vector<MSSpectrum>& annotated,
+    std::vector<MSSpectrum>& picked,
+    std::vector<MSSpectrum>& scored
+  )
+  {
+    std::vector<double> scores(annotated.size());
+    for (UInt i=0; i<annotated.size(); ++i)
+    {
+      double total_tic = 0;
+      for (UInt j=0; j<annotated[i].size(); ++j)
+      {
+        total_tic += annotated[i][j].getIntensity();
+      }
+
+      double avgFWHM = 0;
+      for (UInt j=0; j<picked[i].getFloatDataArrays()[0].size(); ++j)
+      {
+        avgFWHM += picked[i].getFloatDataArrays()[0][j];
+      }
+      avgFWHM /= picked[i].getFloatDataArrays()[0].size();
+
+      SignalToNoiseEstimatorMedian<MSSpectrum> sne;
+      Param p;
+      p.setValue("win_len", 40.0);
+      p.setValue("noise_for_empty_window", 2.0);
+      p.setValue("min_required_elements", 10);
+      sne.setParameters(p);
+      MSSpectrum::const_iterator it;
+      sne.init(annotated[i].begin(),annotated[i].end());
+      double avgSNR = 0.0;
+      for (it=annotated[i].begin(); it!=annotated[i].end(); ++it)
+      {
+        avgSNR += sne.getSignalToNoise(it);
+      }
+      avgSNR /= annotated[i].size();
+
+      double log10_total_tic = log10(total_tic);
+      double one_over_avgFWHM = 1.0/avgFWHM;
+      double score = log10_total_tic + one_over_avgFWHM + avgSNR;
+
+      MSSpectrum spectrum = annotated[i];
+      spectrum.getFloatDataArrays().resize(5); //TODO update this .resize() in case we remove the tic/fwhm/snr infos
+      spectrum.getFloatDataArrays()[1].setName("score");
+      spectrum.getFloatDataArrays()[1].push_back(score);
+      //TODO should we remove the following info?
```

Comment:
I think it will be a good thing to leave them in there.  Could be quite useful for debugging or trying to come up with a suitable set of score weights.